grid2op/multi_agent/multiAgentEnv.py

```diff
@@ -0,0 +1,562 @@
+                
+            * observation (MADict) :
+                - keys : agents' names
+                - values : agent's observation
+            
+            * agent_order : 
+                Gives the order (priority) of agents. If there’s a conflict, the action of the agent with higher
+                priority (smaller index in the list) is kept.
+                
+            * _subgrids (MADIct):
+                - keys : agents’ names
+                - values : created subgrids of type SubGridObjects by _build_subgrids method
+
+                
+            * action_spaces (MADict) :
+                - keys: agents’ names
+                - values : action spaces created from _action_domains and _subgrids
+            
+            * action_spaces (MADict) :
+                - keys: agents’ names
+                - values : observation spaces created from _observation_domains and _subgrids
+                      
+ 
+        """
+        
+        # added to the class name, if you want to build multiple ma environment with the same underlying environment
+        self._add_to_name = _add_to_name  
+        
+        if copy_env:
+            self._cent_env : Environment = env.copy()
+        else:
+            self._cent_env : Environment = env
+        
+        self._verify_domains(action_domains)
+        self._action_domains = {k: {"sub_id": copy.deepcopy(v)} for k,v in action_domains.items()}
+        self.num_agents = len(action_domains)
+        
+        if observation_domains is not None:
+            # user specified an observation domain
+            self._is_global_obs : bool = False
+            if action_domains.keys() != observation_domains.keys():
+                raise("action_domains and observation_domains must have the same agents' names !")
+            self._observation_domains = {k: {"sub_id": copy.deepcopy(v)} for k,v in observation_domains.items()}
+            self._verify_domains(observation_domains)
+        else:
+            # no observation domain specified, so I assume it's a domain
+            # with full observability
+            self._is_global_obs : bool = True
+            self._observation_domains = None
+        
+        self.agents = list(action_domains.keys())
+        self.num_agents = len(self.agents)
+        
+        self.illegal_action_pen = illegal_action_pen
+        self.ambiguous_action_pen = ambiguous_action_pen
+        self.rewards = dict(
+            zip(
+                self.agents, [0. for _ in range(self.num_agents)]
+            )
+        )
+        self._cumulative_rewards = dict(
+            zip(
+                self.agents, [0. for _ in range(self.num_agents)]
+            )
+        )
+        self.observations = dict(
+            zip(
+                self.agents, [None for _ in range(self.num_agents)]
+            )
+        )
+        self.agent_order = AgentSelector(self.agents, agent_order_fn)
```

Comment:
Again: have self.agent_order be a list and set it to `self.agent_order = np.array(sorted(list(self.agents)))` for the v0 it will largely simplify everything for v0.0

You will have more complex implementation with something else later on if you have time. 

Basically: no time for "AgentSelector" here

Reply:
Ok